xwrf/destagger.py

```diff
@@ -0,0 +1,90 @@
+import xarray as xr
+
+
+def _drop_attrs(attrs_dict, attrs_to_drop):
+    try:
+        return {k: v for k, v in attrs_dict.items() if k not in attrs_to_drop}
+    except AttributeError:
+        return None
+
+
+def _destag_variable(datavar, stagger_dim=None, unstag_dim_name=None):
+    """
+    Destaggering function for a single wrf xarray.Variable
+
+    Based off of the wrf-python destagger function
+    https://github.com/NCAR/wrf-python/blob/22fb45c54f5193b849fdff0279445532c1a6c89f/src/wrf/destag.py
+    Copyright 2016 University Corporation for Atmospheric Research, reused with modification
+    according to the terms of the Apache License, Version 2.0
+
+    Parameters
+    ----------
+    datavar : xarray.Variable
+        Data variable to be destaggered
+    stagger_dim : str, optional
+        Name of dimension to unstagger. Defaults to guessing based on name (ends in "_stag")
+    unstag_dim_name : str, option
+        String to which to rename the dimension after destaggering. Example would be
+        "west_east" for "west_east_stag". By default the dimenions will be renamed the text in
+        front of "_stag" from the "stagger_dim" field.
+
+    Returns
+    -------
+    xarray.Variable
+        The destaggered variable with renamed dimension
+    """
+    # get the coordinate to unstagger
+    # option 1) user has provided the dimension
+    if stagger_dim and stagger_dim not in datavar.dims:
+        # check that the user-passed in stag dim is actually in there
+        raise ValueError(f'{stagger_dim} not in {datavar.dims}')
+
+    # option 2) guess the staggered dimension
+    else:
```

Comment:
This is probably not relevant IRL, but it might be nice to have a test case for it.
```suggestion
    elif stagger_dim is None:
```